OpenTrons/opentrons-api
# feat(app, app-shell): add download progress to robot update flows

Closes RAUT-789, RAUT-827



# Overview

If the robot update cache files are still downloading, no longer is an error thrown. Instead, the update modal renders and a downloading stage of the update flow begins. Additionally, release files are downloaded and returned to the browser layer separately from system files, so users may see release notes as soon as they are downloaded.

### What's Changed

The substantive changes are in app-shell/src/robot-update/index & release-files and app/src/redux/robot-update/epic & reducer. I've left comments to help explain what's happening.

### Before

https://github.com/Opentrons/opentrons/assets/64858653/5719968a-97b6-427e-9f58-5636aa1f3706

### After

https://github.com/Opentrons/opentrons/assets/64858653/c3c9d455-b821-4e9a-b142-d8a47325f04c




# Test Plan
- Build the app with make -C app-shell package. Failing the signing step is expected (and fine for testing purposes).
- Before opening the app, clear out the robot-update-cache files (located in Users/username/.Library/Application Support/opentrons/robot-update-cache).
- Perform the next step quickly if you have a good downlink. If you want to throttle your bandwidth, you can use the Network Link Conditioner offered by Mac. 
- Follow the video flow, testing on a Flex. You may need to downgrade a robot first via zip-file (via USB if you're feeling bold).
- Repeat testing flow on an OT-2 (although there should be no functional difference).


# Changelog
- Initiating a robot update while the files are still downloading no longer throws an error. 


# Review requests
- Let me know if you need help testing!


# Risk assessment
Medium - Getting the logic right is important, but it's relatively (keyword) straightforward as far as update flows are concerned...
